### DISCUSSION TITLE
Unable to download and install

### DISCUSSION THREAD

**Discussion by TJHeuvel** [microsoft/openjdk#625] [Q&A] [answered]
When i download the msi or zip files on [this](https://learn.microsoft.com/en-us/java/openjdk/download) page for Windows x64 i get a 0 bytes file. This happens for OpenJDK 21, 17 and 11.

winget also doesnt work here, running the following command:

> winget install Microsoft.OpenJDK.21

Results in the following error:

> 0x8a15002e : Download size does not match expected content length

Is there something going on with the download files?

**Comment by Bulletdev** [comment, marked as answer]
It seems you're encountering issues with downloading the OpenJDK MSI or ZIP files for Windows x64
Possible Causes and Solutions:
1. 
Server Issues: There might be temporary issues with the Microsoft servers hosting the OpenJDK files. This can cause incomplete downloads or mismatched file sizes.
•  Solution: Try downloading the files again after some time. Server issues are often resolved quickly.
2. 
Network Problems: Network interruptions or restrictions could be causing the downloads to fail.
•  Solution: Check your internet connection and ensure there are no network restrictions or firewalls blocking the downloads.
3. 
Corrupted Cache: Sometimes, cached data can interfere with downloads.
•  Solution: Clear your browser cache or try downloading the files using a different browser or network.
4. 
Winget Issues: The error with winget might be due to a mismatch in the expected content length, possibly indicating an issue with the package metadata.
•  Solution: Ensure you have the latest version of winget installed. You can update winget using the following command:

winget upgrade --id Microsoft.Winget.Source   


Alternatively, you can manually download and install the OpenJDK MSI or ZIP files from the Microsoft OpenJDK download page: https://learn.microsoft.com/en-us/java/openjdk/install#install-on-windows-with-the-windows-package-manager-winget
